platform/ios/src/MGLMapboxEvents.m

```diff
@@ -249,6 +239,11 @@ - (NSString *)instanceID {
     return _instanceID;
 }
 
+- (void)userDefaultsDidChange:(NSNotification *)notification {
+    MGLAssertIsMainThread();
```

Comment:
Why the assert here? Could `pauseOrResumeMetricsCollectionIfRequired` not just be dispatched back onto main thread if needed instead?